### PR TITLE
0315 Fix `shared_subscriptions` counting error. followup #12670

### PR DESCRIPTION
**Pull request by JimMoen**
~Fixes <issue-or-jira-number>~
Follow up #12670

Release version: v/e5.6.0

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Added property-based tests for code which performs user input validation
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [x] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [x] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- ~[ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)~
- ~[ ] Change log has been added to `changes/` dir for user-facing artifacts update~

### CODE REVIEW

**Review by zmstone — APPROVED** [review]

File: apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl [hunk 1]
```diff
@@ -112,6 +141,65 @@ t_monitor_current_api_live_connections(_) ->
...
+    _ = emqtt:subscribe(C1, {ShareT, 1}),
+    ok = AssertFun(2),
+
+    _ = emqtt:unsubscribe(C1, ShareT),
+    ok = AssertFun(1),
+    _ = emqtt:subscribe(C1, {ShareT, 1}),
+    ok = AssertFun(2),
+
+    ok = emqtt:disconnect(C),
+    %% C1 clean_start = false
```
Comment (qzhuyan) [inline apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl:192]: 1. L181 `C1` clean start is `true`, do you mean `C` ?

2. clean_start flag is used for new connection. 
If you mean session persist after disconnect, it should be "session expire interval".

File: apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl [hunk 2]
```diff
@@ -112,6 +141,65 @@ t_monitor_current_api_live_connections(_) ->
...
+    %% means share-subscription is still alive
+    ok = AssertFun(2),
```
Comment (qzhuyan) [inline apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl:194]: 🤔  Strange it is still 2. 
What makes `C` session persistent after disconnect?

File: apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl (hunk 1 above)
Comment (JimMoen) [reply apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl:192]: 1. My fault， `C` of course
2. The default proto ver is `3.1.1`, `clean_start` enough ?

File: apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl (hunk 2 above)
Comment (JimMoen) [reply apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl:194]: `C` used proto `3.1.1`  with `clean_start=false`, 
> aka `clean_session=false` in mqtt_v3.1.1

**Review by HJianBo — APPROVED** [review]

**Pull request merged**